Develop v1

Hendrik040 opened the pull request:
CodeRabbit review


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

# Release Notes

* **New Features**
  * Messaging system for professor-student communication with reply threads and read status tracking
  * Cohort invitation system with shareable invite links
  * PDF case study upload and grading materials management
  * Real-time progress tracking for PDF parsing
  * Rubric-based grading configuration for scenarios
  * Docker containerization for simplified deployment
  * Enhanced simulation scenes with intro messages and adaptive persona guidance

* **Infrastructure**
  * Database connection pool optimization
  * Improved error handling and logging across the platform
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Hendrik040 commented on backend/api/messages.py:
```diff
@@ -0,0 +1,468 @@
+    db.commit()
+    db.refresh(message)
+    
+    # Create notification for the recipient
+    if current_user.role == 'professor' and recipient.role == 'student':
+        create_professor_message_notification(
+            db=db,
+            professor=current_user,
+            student=recipient,
+            message_subject=message_data.subject,
+            cohort_id=message_data.cohort_id
+        )
+    elif current_user.role == 'student' and recipient.role == 'professor':
+        # For new messages from students to professors, use a different notification type
+        notification_service.create_notification(
+            db=db,
+            user_id=recipient.id,
+            notification_type='student_message',
+            variables={
+                'student_name': current_user.full_name,
+                'message_subject': message_data.subject
+            },
+            data={
+                'student_id': current_user.id,
+                'cohort_id': message_data.cohort_id,
+                'message_type': 'student_message'
+            }
+        )
+    
```
@coderabbitai plan

Sbcoit committed:
Merge pull request #68 from Hendrik040/ECS-39-Simulation-Builder-Autofill-Not-Working

fix: use query params for drafts instead of separate endpoint to avoi…

Sbcoit committed:
chore: force rebuild to clear cache

Sbcoit committed:
chore: add build timestamp to force cache bust

Sbcoit committed:
fix: proxy automatically adds trailing slashes for FastAPI endpoints

Sbcoit committed:
fix: proxy now follows redirects for GET requests

Sbcoit committed:
fix: correct delete scenario endpoint path

altran03 committed:
add initial grading configuration + polish scenario publish status w/new migration

Sbcoit committed:
fix: filter out soft-deleted scenarios from GET endpoints

altran03 committed:
Merge pull request #69 from altran03/alvin/ECS-40

ECS40-Configurable Grading Agent

Sbcoit committed:
refactor: remove PyPDF2 fallback, use LlamaParse only

Sbcoit committed:
update reqs

Sbcoit committed:
Merge pull request #71 from Hendrik040/ECS-39-Simulation-Builder-Autofill-Not-Working

Ecs 39 simulation builder autofill not working

altran03 committed:
Merge pull request #70 from Hendrik040/ECS-40-Configurable-Grading-Agent

Ecs 40 configurable grading agent

altran03 committed:
migration fix

altran03 committed:
Merge pull request #72 from altran03/ECS-40/migration_fix

migration fix

altran03 committed:
Fix database schema: add grading_config fields to ScenarioPublishingResponse

- Added grading_config_completed and grading_config fields to ScenarioPublishingResponse schema
- This resolves the API error where the response model was missing fields that the database query was trying to select
- Fixes the 'column scenarios.grading_config_completed does not exist' error in production

altran03 committed:
Merge pull request #73 from altran03/ECS-40/migration_fix

schema fix

altran03 committed:
Created new migration da4c671ffd49 to add grading_config and grading_config_completed columns

altran03 committed:
Merge pull request #74 from altran03/ECS-40/migration_fix

Created new migration da4c671ffd49

altran03 committed:
rebase, update saving+loading

altran03 committed:
Merge pull request #75 from altran03/ECS-40-Configurable-Grading-Agent

Ecs 40 configurable grading agent

altran03 committed:
Merge pull request #76 from Hendrik040/ECS-40-Configurable-Grading-Agent

Ecs 40 configurable grading agent

Sbcoit committed:
show only user generated

Sbcoit committed:
fix tag status

Sbcoit committed:
status fix

Sbcoit committed:
test fixes

Sbcoit committed:
done

Sbcoit committed:
Merge pull request #77 from Hendrik040/ECS-47-Dashboard-Displays-Other-User-s-Simulations

Ecs 47 dashboard displays other user s simulations

Sbcoit committed:
done

Sbcoit committed:
Merge pull request #78 from Hendrik040/ECS-47-Dashboard-Displays-Other-User-s-Simulations

Ecs 47 dashboard displays other user s simulations

Sbcoit committed:
Adding functionality to student cohort page

Sbcoit committed:
added status logic

Sbcoit committed:
updated endpoints and ui

Sbcoit committed:
Student can now play simulation

Sbcoit committed:
working assigned

Sbcoit committed:
Merge pull request #79 from Hendrik040/ECS-47-Dashboard-Displays-Other-User-s-Simulations

PR here because i forgot to switch to the correct branch for Cohort Infrastructure

Sbcoit committed:
update database

Sbcoit committed:
fixes to alembic migrations

Sbcoit committed:
update start command

Sbcoit committed:
check student auth

Sbcoit committed:
test fixes

Sbcoit committed:
test fixes

Sbcoit committed:
fixes to db

Sbcoit committed:
test db fixes

Sbcoit committed:
test db fixes

Sbcoit committed:
test db fixes

Sbcoit committed:
test features

Sbcoit committed:
test fixes

Sbcoit committed:
test done:

Sbcoit committed:
Merge pull request #80 from Hendrik040/ECS-41-Cohort-Infrastructure

Ecs 41 cohort basic infrastructure

Sbcoit committed:
small fix to scene numbers

Sbcoit committed:
Merge pull request #81 from Hendrik040/ECS-41-Cohort-Infrastructure

small fix to scene numbers

Sbcoit committed:
fix images not loading

Sbcoit committed:
Merge pull request #82 from Hendrik040/ECS-41-Cohort-Infrastructure

fix images not loading

Sbcoit committed:
small fixes

Sbcoit committed:
Merge pull request #83 from Hendrik040/ECS-41-Cohort-Infrastructure

small fixes

Sbcoit committed:
done

Sbcoit committed:
test fixes

Sbcoit committed:
Merge pull request #84 from Hendrik040/ECS-49-Implement-Typing-Effect-and-Direct-Mentions-Message

Ecs 49 implement typing effect and direct mentions message

Sbcoit committed:
make student dashboard functioning

Sbcoit committed:
notifcations number

Sbcoit committed:
invite dropdown QOL

Sbcoit committed:
Merge pull request #85 from Hendrik040/ECS-49-Implement-Typing-Effect-and-Direct-Mentions-Message

Ecs 49 implement typing effect and direct mentions message

altran03 committed:
initial tabular simulation page (grading tab is still just a template)

altran03 committed:
revert img handling

altran03 committed:
file uploads

Sbcoit committed:
Fix state handling for drafts

Sbcoit committed:
Merge pull request #86 from Hendrik040/ECS-49-Implement-Typing-Effect-and-Direct-Mentions-Message

Fix state handling for drafts

Sbcoit committed:
images load properly now

Sbcoit committed:
Merge pull request #87 from Hendrik040/ECS-49-Implement-Typing-Effect-and-Direct-Mentions-Message

images load properly now

hendrik-devRel commented on README.md:
```diff
@@ -36,33 +44,86 @@ pip install -r requirements.txt
 - **API Documentation**: http://localhost:8000/docs
+- **Database**: localhost:5432 (username/password)
+- **Redis**: localhost:6379
+
```
@coderabbitai please reverse engineer the requirements for this function from the code

altran03 committed:
implement rag + chunking for grading uploads

Sbcoit committed:
finished implementing system_prompt and tweaked langchain

altran03 committed:
update grading endpoint to use RAG agent

Sbcoit committed:
implemented system-prompt as well fixed langchain and pgvector

Sbcoit committed:
Merge pull request #90 from Hendrik040/ECS-48-Persona-Creation-with-Advanced-Mode

Ecs 48 persona creation with advanced mode

Sbcoit committed:
deleted unneeded files

Sbcoit committed:
Merge pull request #91 from Hendrik040/ECS-48-Persona-Creation-with-Advanced-Mode

deleted unneeded files

altran03 committed:
feature/rubric

altran03 committed:
update grading context

altran03 committed:
update grading strictness

Sbcoit committed:
testing new fixes

Hendrik040 committed:
Update .coderabbit.yaml

adding the full blown yaml file

Hendrik040 committed:
Merge pull request #92 from Hendrik040/feature/configure-codeRabbit-yaml

Update .coderabbit.yaml

Sbcoit committed:
fixed persona system_prompt memory leak

Sbcoit committed:
Merge origin changes while preserving persona isolation fixes

Sbcoit committed:
Merge pull request #95 from Hendrik040/ECS-48-Persona-Creation-with-Advanced-Mode

Ecs 48 persona creation with advanced mode

Sbcoit committed:
still need fix timeout logic

Sbcoit committed:
finalize fixes

Sbcoit committed:
Merge pull request #96 from Hendrik040/ECS-48-Persona-Creation-with-Advanced-Mode

Ecs 48 persona creation with advanced mode

Sbcoit committed:
migration issue:

Sbcoit committed:
db fixes

Sbcoit committed:
Merge pull request #97 from Hendrik040/ECS-48-Persona-Creation-with-Advanced-Mode

Ecs 48 persona creation with advanced mode

Hendrik040 committed:
added a new coderabbit.yaml file with uopdated configs that are more taylored towards what we need

Hendrik040 committed:
added the base-branch extansion '.*' which should force coderabbit to review all branches  PRs

Hendrik040 committed:
fixed small syntax error in coderabbit yaml all branch config was not set as an array

Hendrik040 committed:
Merge pull request #98 from Hendrik040/update-coderabbit-yaml

Update coderabbit yaml

Summary by CodeRabbit
New Features
Added configurable file pattern matching for code generation settings, enabling targeted file specification using glob patterns for custom configuration files.

Hendrik040 committed:
Update README.md

Sbcoit committed:
implemented cleaner persona responses

Sbcoit committed:
touch up sidebar

altran03 committed:
progress clarity

altran03 committed:
Fix: Remove gradingConfig reference after rebase

altran03 committed:
Fix TypeScript errors in rubric configuration

altran03 committed:
Fix migration conflicts after rebase - merge heads and remove table drops

altran03 committed:
Merge pull request #102 from Hendrik040/ECS-40.01-Grading-Page

Ecs 40.01 - RAG and Rubric

Hendrik040 committed:
changed CI/CD pipeline

Commit:
just changing the visivility of the chat bar here

Hendrik040 committed:
Merge pull request #103 from Hendrik040/ECS-57-remove-hide-the-chat-page-on-the-student-interface

Ecs 57 remove hide the chat page on the student interface

Hendrik040 committed:
Merge pull request #104 from Hendrik040/staging

Staging

Commit:
hiddig the google log in flow for now

Commit:
Merge branch 'ECS-63-hidding-the-google-login-register-for-now' of https://github.com/Hendrik040/n-aible_EdTech_Sims into ECS-63-hidding-the-google-login-register-for-now

Hendrik040 committed:
Merge pull request #105 from Hendrik040/ECS-63-hidding-the-google-login-register-for-now

Ecs 63 hidding the google login register for now

Sbcoit committed:
finished new sim interface

Sbcoit committed:
Merge pull request #107 from Hendrik040/ECS-50-New-Simulation-User-Interface

Ecs 50 new simulation user interface

Sbcoit committed:
fix db issues test

Sbcoit committed:
Merge pull request #108 from Hendrik040/ECS-50-New-Simulation-User-Interface

fix db issues test

Hendrik040 committed:
Merge pull request #100 from Hendrik040/develop-v1

Develop v1 pushing to production

Sbcoit committed:
update to frontend

Sbcoit committed:
persona pfp implemented

Sbcoit committed:
reached limit

Sbcoit committed:
Merge pull request #109 from Hendrik040/ECS-50-New-Simulation-User-Interface

Ecs 50 new simulation user interface

Sbcoit committed:
new ui

Sbcoit committed:
Merge pull request #110 from Hendrik040/ECS-50-New-Simulation-User-Interface

new ui

Sbcoit committed:
small ui fix

Commit:
hidding google Oauth since it's not working yet

Hendrik040 committed:
Merge pull request #106 from Hendrik040/ECS-63-hidding-the-google-login-register-for-now

Ecs 63 hidding the google login register for now

Hendrik040 committed:
Update frontend/components/MessagingModal.tsx

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

Hendrik040 committed:
Update frontend/components/RoleChooser.tsx

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

Hendrik040 committed:
Change review profile from 'assertive' to 'chill'

changing the strictness of CodeRabbit to chill because it's too noisy otherwise

Hendrik040 committed:
Merge branch 'staging' into develop-v1

Hendrik040 committed:
Merge branch 'production' into develop-v1

Commit:
fixed syntanx error

Hendrik040 committed:
Merge pull request #112 from Hendrik040/but/small-tsc-syntax-error

fixed syntanx error

Hendrik040 committed:
Merge pull request #111 from Hendrik040/develop-v1

Develop v1

Sbcoit committed:
fix position of mModal

Sbcoit committed:
Merge origin/develop-v1: Resolve conflict in MessagingModal.tsx, combine portal approach with accessibility improvements

Sbcoit committed:
fix tag

Sbcoit committed:
small timeline interface fix

Sbcoit committed:
Improved grading form

Sbcoit committed:
update logo

Sbcoit committed:
fix logo

Sbcoit committed:
finished invite links

Sbcoit committed:
Merge pull request #113 from Hendrik040/ECS-64-Make-invite-link-function

finished invite links

Sbcoit committed:
fix cohort not creating

Sbcoit committed:
improved loading hopefully

Sbcoit committed:
fix small bugs

Commit:
addind a branch with unit tests

Commit:
added S3 base services

Sbcoit committed:
login fix

Commit:
ECS-66 Implement S3

Sbcoit committed:
some more login warning message tweaks

Sbcoit committed:
Merge pull request #114 from Hendrik040/ECS-65-Invite-Link-Signup-Process-broken

Ecs 65 invite link signup process broken

Sbcoit committed:
finished

Sbcoit committed:
Merge pull request #116 from Hendrik040/ECS-65-Invite-Link-Signup-Process-broken

autofill forms no longer are lost on page reload or exit

Sbcoit committed:
fixed

Sbcoit committed:
Merge pull request #117 from Hendrik040/ECS-60-Enhance-the-context-of-the-agent

fixed

Sbcoit committed:
fix not correct error message not showing up

Sbcoit committed:
Merge pull request #118 from Hendrik040/ECS-60-Enhance-the-context-of-the-agent

fix not correct error message not showing up

Hendrik040 committed:
Merge pull request #101 from Hendrik040/develop-v1

pushing Develop v1 to staging

Sbcoit committed:
small fix to save

Commit:
implemented s3 aws successful

Sbcoit committed:
implemented more auto-save logic

Commit:
stable S3 bucket upload - still missing the correct bucket hirachy but it's working

Commit:
S3 final configs to save PDF correctly

Hendrik040 committed:
Merge pull request #115 from Hendrik040/ECS-66-implement-s-3

Ecs 66 implement s 3

Sbcoit committed:
Merge develop-v1 into ECS-68-Make-grades-editable-by-professor

Sbcoit committed:
Merge pull request #122 from Hendrik040/ECS-68-Make-grades-editable-by-professor

Ecs 68 make grades editable by professor

Sbcoit committed:
fix cookies

Sbcoit committed:
remove icon

Commit:
commit the first fix for the laoding issue

Hendrik040 committed:
Merge remote-tracking branch 'origin/develop-v1' into ECS-66-implement-s-3

Hendrik040 committed:
Merge pull request #124 from Hendrik040/ECS-66-implement-s-3

Hendrik040 committed:
proud to be solving this issue with a fix finally (hopefully)

Hendrik040 committed:
Merge pull request #126 from Hendrik040/ECS-66-implement-s-3

proud to be solving this issue with a fix finally (hopefully)

Sbcoit committed:
fixes to new domain

Sbcoit committed:
changed domain causing issues

Sbcoit committed:
fix cookie simulation issue

Sbcoit committed:
Merge pull request #127 from Hendrik040/ECS-68-Make-grades-editable-by-professor

Ecs 68 make grades editable by professor

Sbcoit committed:
smooth slider

Hendrik040 committed:
fix endpoint publishing

Hendrik040 committed:
Merge pull request #128 from Hendrik040/ECS-66-implement-s-3

fix endpoint publishing

Sbcoit committed:
done

Sbcoit committed:
Merge pull request #129 from Hendrik040/ECS-68-Make-grades-editable-by-professor

Ecs 68 make grades editable by professor

Sbcoit committed:
fix build fail

Sbcoit committed:
Merge pull request #130 from Hendrik040/ECS-68-Make-grades-editable-by-professor

fix build fail

Hendrik040 committed:
addind one more fix to this

Hendrik040 committed:
Merge pull request #131 from Hendrik040/ECS-66-implement-s-3

addind one more fix to this

Hendrik040 committed:
fixing

Sbcoit committed:
smal UI changes

Sbcoit committed:
Merge pull request #133 from Hendrik040/ECS-68-Make-grades-editable-by-professor

smal UI changes

Hendrik040 committed:
merge

Hendrik040 committed:
updated fix for the autosave disruption

Hendrik040 committed:
status upload blocked to save draft while processing

Hendrik040 committed:
Merge pull request #134 from Hendrik040/bug/cookie-issue

Bug/cookie issue

Hendrik040 committed:
removed wasabi references

Hendrik040 committed:
fixing breaking things and fixing again ahaha

Hendrik040 committed:
Merge pull request #136 from Hendrik040/bug/dev-server-crash-2

Bug/dev server crash 2

Hendrik040 committed:
updateding deployment changes

Hendrik040 committed:
change deployment stuff

Hendrik040 committed:
Merge pull request #137 from Hendrik040/bug/dev-server-crash-2

Bug/dev server crash 2

Hendrik040 committed:
fixing the llamaparse

Hendrik040 committed:
Merge pull request #139 from Hendrik040/bug/dev-server-crash-2

fixing the llamaparse

Hendrik040 committed:
Merge pull request #138 from Hendrik040/develop-v1

Develop v1

Sbcoit committed:
done just need testing

Sbcoit committed:
should fix saving to S3 Bucket causing crashes

Sbcoit committed:
unique colors

altran03 committed:
dirty temp reset password

altran03 committed:
Merge pull request #141 from Hendrik040/ECS-69-Forgot-my-password-feature-doesn-t-work

dirty temp reset password

altran03 committed:
edit cohort feature

Hendrik040 committed:
added debugging logic

altran03 committed:
Merge pull request #144 from Hendrik040/ECS-73-Add-edit-button-for-Cohort

edit cohort feature

Hendrik040 committed:
Merge remote-tracking branch 'origin/develop-v1' into ECS-70-Messaging-all-agents-using-all-doesn-t-work

altran03 committed:
profilepage

altran03 committed:
Merge pull request #145 from Hendrik040/ECS-81-add-profile-page

profilepage

Sbcoit committed:
Merge pull request #140 from Hendrik040/ECS-70-Messaging-all-agents-using-all-doesn-t-work

Ecs 70 messaging all agents using all doesnt work

Sbcoit committed:
Fixed S3 Saving Issue hopefully

Hendrik040 committed:
Merge pull request #146 from Hendrik040/develop-v1

Develop v1

Hendrik040 committed:
building a fix for the scenario settings

Sbcoit committed:
done

Sbcoit committed:
Merge pull request #147 from Hendrik040/ECS-91-Display-uploaded-case-study-in-the-case-study-tab-and-fix-saving-to-S3-Bucket-causing-crashes

Ecs 91 display uploaded case study in the case study tab and fix saving to s3 bucket causing crashes

Sbcoit committed:
fix proxy issue

Hendrik040 committed:
added minor fix from CR

Hendrik040 committed:
Merge pull request #148 from Hendrik040/bug/fixing-scenario-count

building a fix for the scenario settings

Sbcoit committed:
fix to login

Sbcoit committed:
fix backend not responding

Sbcoit committed:
fix migrations

Sbcoit committed:
fix autofill not working

Sbcoit committed:
the issue was the database, everything works, its just the migrations

Sbcoit committed:
Merge branch 'develop-v1' into ECS-91-Display-uploaded-case-study-in-the-case-study-tab-and-fix-saving-to-S3-Bucket-causing-crashes

Sbcoit committed:
Merge pull request #149 from Hendrik040/ECS-91-Display-uploaded-case-study-in-the-case-study-tab-and-fix-saving-to-S3-Bucket-causing-crashes

the issue was the database, everything works, its just the migrations

Sbcoit committed:
fix losing scenes

Sbcoit committed:
Merge develop-v1: Reverted pull changes, keeping scene deletion fixes

Sbcoit committed:
Merge pull request #151 from Hendrik040/ECS-91-Display-uploaded-case-study-in-the-case-study-tab-and-fix-saving-to-S3-Bucket-causing-crashes

Ecs 91 display uploaded case study in the case study tab and fix saving to s3 bucket causing crashes

Sbcoit committed:
fix persona pfp not showing

Sbcoit committed:
Merge pull request #152 from Hendrik040/ECS-91-Display-uploaded-case-study-in-the-case-study-tab-and-fix-saving-to-S3-Bucket-causing-crashes

fix persona pfp not showing

Sbcoit committed:
Merge branch 'staging' into develop-v1

Sbcoit committed:
Merge pull request #153 from Hendrik040/develop-v1

dev-v1

Sbcoit committed:
test fixes

Sbcoit committed:
test new connnection limit

Sbcoit committed:
fix concurrent requests

Sbcoit committed:
increase cookie timeout

Sbcoit committed:
Merge pull request #154 from Hendrik040/staging

update dev-v1

Sbcoit committed:
fix cookie issue for logout

Sbcoit committed:
Merge pull request #155 from Hendrik040/staging

fix cookie issue for logout

Sbcoit committed:
try safari browser fixes

Sbcoit committed:
fix profiles not loading

Sbcoit committed:
test fixes

Sbcoit committed:
Merge pull request #156 from Hendrik040/ECS-93-Make-persona-profile-pictures-show-up-more-consistently

Ecs 93 make persona profile pictures show up more consistently

Sbcoit committed:
increase connection limit

Sbcoit committed:
increase connect limit

Sbcoit committed:
Merge pull request #158 from Hendrik040/ECS-93-Make-persona-profile-pictures-show-up-more-consistently

Ecs 93 make persona profile pictures show up more consistently
